custom_openstack_operator.md

```diff
@@ -0,0 +1,139 @@
+prerequisites above.  You can view the corresponding progress in the `Actions` tab of your fork:
+
+![Building images](images/github_action_building_images.png)
+![Building images 2](images/github_action_building_images2.png)
+
+### 2. Modify OpenStack Operator to use your custom service operator changes
+
+Once the GitHub actions have finished building your service operator images, you can modify your
+OpenStack operator's `go.mod` files so that they use your fork and branch.  This will also allow
+your custom service operators images to be used with your custom OpenStack operator when you build
+it in a few moments.
+
+1. Use `go get` to acquire the version ID of your service operator's fork and branch
+
+```bash
+$ cd openstack-operator
+$ go get github.com/abays/keystone-operator/api@some_changes
+go: downloading github.com/abays/keystone-operator v0.0.2-0.20241024164030-114f7d6488b9
+go: downloading github.com/abays/keystone-operator/api v0.0.0-20241024164030-114f7d6488b9
+```
+
+Notice the format of the URI: `github.com/<your GitHub user>/<your service operator fork>/api@<your branch with your changes>`.
+Change this to suit your purposes.
+
+Also notice the returned version ID: `vX.X.X-<timestamp>-<commit ID>`
+
+2. Modify your OpenStack operator `go.mod` files (both at the root dir and the `apis` dir) to include a
+`replace` that uses your custom service operator version (do this via your IDE, the output below just
+shows the aftermath)
+
+```bash
+$ tail -1 go.mod
+replace github.com/openstack-k8s-operators/keystone-operator/api => github.com/abays/keystone-operator/api v0.0.0-20241024164030-114f7d6488b9
+$
+$ tail -1 apis/go.mod
+replace github.com/openstack-k8s-operators/keystone-operator/api => github.com/abays/keystone-operator/api v0.0.0-20241024164030-114f7d6488b9
+```
+
+### 3. Build your custom OpenStack Operator
+
+1. If your GitHub username is NOT the same as your Quay username, you will need to add the following
+hack to your OpenStack operator's `hack/pin-bundle-images.sh`:
+
+```bash
+$ git diff hack/pin-bundle-images.sh
+diff --git a/hack/pin-bundle-images.sh b/hack/pin-bundle-images.sh
+index 57b35d7..eb47338 100755
+--- a/hack/pin-bundle-images.sh
++++ b/hack/pin-bundle-images.sh
+@@ -40,6 +40,9 @@ for MOD_PATH in ${MOD_PATHS}; do
+     fi
+ 
+     GITHUB_USER=$(echo $MOD_PATH | sed -e 's|github.com/\(.*\)/.*-operator/.*$|\1|')
++    if [ "$GITHUB_USER" == "<your GitHub username>" ]; then
++        GITHUB_USER=<your Quay username>
++    fi
+     CURL_REGISTRY="quay.io"
+     REPO_CURL_URL="https://${CURL_REGISTRY}/api/v1/repository/openstack-k8s-operators"
+     REPO_URL="${CURL_REGISTRY}/openstack-k8s-operators"
+```
+
+2. Build the custom OpenStack operator (this will take a few minutes)
+
+```bash
+$ IMAGE_TAG_BASE=quay.io/andrewbays/openstack-operator VERSION=0.0.1 IMG=$IMAGE_TAG_BASE:v$VERSION make manifests build docker-build docker-push bundle bundle-build bundle-push catalog-build catalog-push
```

Comment:
if the user is different you just have to specify `IMAGENAMESPACE=<your quay user>` in addition to the `IMAGE_TAG_BASE`. then step 1 is not required.